meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch

```diff
@@ -0,0 +1,169 @@
+From 5a13a97c2112ec58d5171b8cbbe4c85e8570c56d Mon Sep 17 00:00:00 2001
+From: Herrie <Github.com@herrie.org>
+Date: Thu, 21 Dec 2023 12:54:37 +0100
+Subject: [PATCH] com.webos.service.mediacontroller: Make it work generically
+
+Not liking the hardcoded bits from OSE, let's make it more flexible and keep the same functionality
+---
+Upstream-Status: Pending
+
+ CMakeLists.txt              |  6 ++----
+ src/MediaControlService.cpp | 35 ++++++++++++++---------------------
+ src/MediaSessionManager.cpp |  2 +-
+ 3 files changed, 17 insertions(+), 26 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 13f158f..6434b6e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -45,12 +45,10 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+ add_definitions(-DUSE_TEST_METHOD)
+ endif()
+ 
+-if(WEBOS_TARGET_MACHINE STREQUAL "raspberrypi4" OR WEBOS_TARGET_MACHINE STREQUAL "raspberrypi4-64" OR WEBOS_TARGET_MACHINE STREQUAL "qemux86" OR WEBOS_TARGET_MACHINE STREQUAL "qemux86-64")
+-add_definitions(-DPLATFORM_RASPBERRYPI4)
+-endif()
+-
+ if(WEBOS_TARGET_MACHINE STREQUAL "sa8155")
+ add_definitions(-DPLATFORM_SA8155)
++else ()
++add_definitions(-DPLATFORM_GENERIC)
+ endif()
+ 
+ webos_add_compiler_flags(ALL -Wall -funwind-tables)
+diff --git a/src/MediaControlService.cpp b/src/MediaControlService.cpp
+index 5faecac..537c8d8 100644
+--- a/src/MediaControlService.cpp
++++ b/src/MediaControlService.cpp
+@@ -150,14 +150,7 @@ bool MediaControlService::onBTAdapterQueryCb(LSHandle *lshandle, LSMessage *mess
+       std::string adapterAddress = adapters[i]["adapterAddress"].asString();
+       PMLOG_INFO(CONST_MODULE_MCS,"%s : adapterName : %s, adapterAddress : %s",
+                                    __FUNCTION__, adapterName.c_str(), adapterAddress.c_str());
+-
+-#if defined(PLATFORM_RASPBERRYPI4)
+-      if(("raspberrypi4 #2" == adapterName) || ("raspberrypi4" == adapterName) ||
+-         ("raspberrypi4-64 #2" == adapterName) || ("raspberrypi4-64" == adapterName) ||
+-         ("qemux86-64 #2" == adapterName) || ("qemux86-64" == adapterName)) {
+-#elif defined(PLATFORM_SA8155)
+-      if ("sa8155 Bluetooth hci0" == adapterName) {
+-#endif
++      if (("" != adapterName) && ("sa8155 Bluetooth hci1" != adapterName)) {
```

Comment:
@Tofee Not 100% sure about this "solution" here. Happy for suggestions for something else.

Reply:
To stay a bit closer to the original version, you could write something like this:
```c
#if defined(PLATFORM_GENERIC)
      if (!adapterName.empty()) {
#elif defined(PLATFORM_SA8155)
      if ("sa8155 Bluetooth hci0" == adapterName) {
#endif
```

Reply:
I'm more worried if it does what it's intend to do. I don't really understand what they were trying to do ;) 

On pinephonepro the adaptername shows up as "pinephonepro" which I guess is what we expect
`Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlService {} onBTAdapterQueryCb : adapterName : pinephonepro, adapterAddress : aa:aa:aa:aa:aa:aa
Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlService {} onBTAdapterQueryCb : payload = {"adapterAddress":"aa:aa:aa:aa:aa:aa","subscribe":true} for first adapter
Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlService {} onBTAdapterQueryCb : adapterAddress = aa:aa:aa:aa:aa:aa deviceSetId = RSE-L [182]
Dec 22 10:07:53 pinephonepro com.webos.service.mediacontroller[1624]: [] [pmlog] mediacontroller MediaControlPrivate {}  setBTAdapterInfo
`

Reply:
I guess they just want to make sure we go into that "if" only for the adapter we build for. They do like to have per-target binaries, though I don't really see the big advantage of doing that here.